go/test/endtoend/sharding/base_sharding.go

```diff
@@ -355,3 +389,128 @@ func CheckTabletQueryService(t *testing.T, vttablet cluster.Vttablet, expectedSt
 		assert.Equal(t, tabletStatus, expectedStatus)
 	}
 }
+
+// CheckShardQueryServices checks DisableQueryService in the shard record's TabletControlMap.
```

Comment:
for all shards? That should be in the comment, otherwise this is the same as the comment for the next function.